### DISCUSSION TITLE
Regolith in Debian officially

### DISCUSSION THREAD

**Discussion by jmtd** [regolith-linux/regolith-desktop#943] [Q&A] [answered]
Is there any interest or desire on adding Regolith to Debian as official packages? Asking as a (somewhat lay) Debian developer. Thanks

**Comment by kgilmer** [comment, marked as answer]
Hi @jmtd , wow what timing!  I am just back from a conference in which that very topic was discussed.  I'm now getting setup as a Debian maintainer in preparation to push packages to the official places.  I have a few Debian developers helping me out at the moment.

**Reply by jmtd** [reply]
Great to hear. I'll keep an eye on debian-devel and debian-mentors and if there's something I think I can help with on the Debian side I might be able to pitch in!

**Comment by Whyglobaleyes** [comment]
Hi, 
It's of no bearing because I can't do anything useful to help but "I'd like to second that emotion" because I'd be very, very happy to see an official debian regolith image. 

First used regolith 1.6 on ubuntu base but jumped to debian several years ago. (Shamefaced confession: my main reason for switching is utterly trivial - I didn't like the snap processes messing up my htop and lsblk outputs.) I've never quite gotten regolith running quite so smoothly on debian as I did on ubuntu hence the thumbs up here.

bw

**Comment by MrWazaby** [comment]
Hello! Do you have an update on this?
I think that it will be great to have the possibility to use Regolith natively on Debian!

**Comment by trharmon0** [comment]
I'm tagging on on this because it is, admittedly, not a perfect match as a question but it's the closets I've found...

I would really like to run Debian unstable as a rolling release on my workstation. The thing holding me back is my reliance on regolith's i3 which is not available in unstable. 

Are their resources for how to get started hacking on regolith? Specifically for Debian and Debian variants?

**Reply by kgilmer** [reply]
Hi @trharmon0 From the 3.x series I believe Regolith uses upstream's package of `i3`.  In earlier major versions we packaged it because the upstream packages were very behind, but that was addressed and now Regolith relies on upstream for both Debian and Ubuntu.  Here's where i3 is coming from on my machine:

```
~$ apt list --installed | grep i3

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

i3-wm/plucky,now 4.24-1 arm64 [installed,automatic]
```

Regarding running Regolith on `sid`, I have noticed failures, as expected, as new packages are added to the Debian repo sthat introduce breaking changes in other packages, or when packages are removed or replaced entirely.  There are probably ways in dealing with these kinds of changes, but have found for myself that running `testing` is a more reliable experience.

Are you seeing a particular error when running Regolith on `sid`?

**Reply by trharmon0** [reply]
I have not attempted it at all recently. The last time I tried it with vanilla Sid was a few years ago. I want to move to a rolling release of Sid via Siduction and so I thought I might try to bring Regolith along with me. If I were able to get R3, rolling Sid, and BTRFS snapshotting ... well, now I'm on to something!

I've been browsing the GH space and just trying to figure where I would even start. It might just be to get a Siduction VM (check) and then apply the R3 apt lines for bookworm and see where that leaves me.

If you have thoughts on that idea...

**Reply by kgilmer** [reply]
There is an apt repo line for Debian Testing: `testing` that will be more current than `bookworm`.  FWIW there are UI refresh issues that occur when Regolith is run in a VM that are not present when running on "bare metal".  More information about supported installation targets available here: https://regolith-desktop.com/docs/using-regolith/install/

**Reply by trharmon0** [reply]
I'll give testing apt lines a try on Sid/Siduction and see how it goes. I'm already quite a ways out into the wilderness by running  Siduction so mixing testing and unstable apt lines is not all that more craziness. :D If there's anything interesting to share I will do so in Discussions.

thx
